runtime/compiler/trj9/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -15688,26 +15688,19 @@ J9::Z::TreeEvaluator::BCDCHKEvaluatorImpl(TR::Node * node,
 
    if(isResultPD)
       {
+      TR::Register* srcBaseReg = cg->evaluate(secondChild);
+      TR::MemoryReference* srcMR = generateS390MemoryReference(srcBaseReg, 0, cg);
```

Comment:
Can we fuse these two lines to be:

```
generateS390MemoryReference(secondChild, cg, false);
```

Reply:
This `secondChild` is the second child of BCDCHK and is an `aladd` node. It's not under a load or store node. Hence, it can not use the `generateS390MemoryReference(TR::Node*,TR::CodeGenerator*,bool)` API.  This particular API seems to be designed for load/store nodes with SymbolReferences. It calls a `MemoryReference` constructor that only sets its own symbolReference field if the root node has the `hasSymbolRef` property. `aladd` node has no such property.

 If we do `generateS390MemoryReference(secondChild, cg, false)`, it'll likely crash in the `MemoryReference::calcDisplacement()` method due to NULL SymbolReference pointer.

Reply:
Besides, this two lines are used to generate packed decimal data copy back in the BCDCHK OOL path. It's not on the performance critical path so whether the address is folded or not makes little difference in terms of performance.

Reply:
Ok, in that case I'd like to see another PR in OMR documenting the semantics of `generateS390MemoryReference(TR::Node*,TR::CodeGenerator*,bool)` and what it should be used for as this is a subtlety that is non-obvious.